torch/distributed/_composable/fsdp/fully_shard.py

```diff
@@ -0,0 +1,44 @@
+from typing import Any
+
+import torch.nn as nn
+
+from torch.distributed._composable import contract
+from torch.distributed._composable_state import _insert_module_state
+
+
+@contract()
+def fully_shard(
```

Comment:
should definitely apidoc this one

Reply:
Does it need to be done in this PR? The `fully_shard` function is barebones here and does not really do anything. Since this is not a public API, I was thinking to defer the docstring to when there is actually some semblance of FSDP being implemented.

Reply:
I am also happy to fill in the docstring here with some description first though.

Reply:
yea thats fine